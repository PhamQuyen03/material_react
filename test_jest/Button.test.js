import React from 'react';
import Button from '../src/Button.react';
import renderer from 'react-test-renderer';

test('Test show table', () => {
  const component = renderer.create(
    <Button >Search</Button>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // tree.props.onMouseEnter();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});