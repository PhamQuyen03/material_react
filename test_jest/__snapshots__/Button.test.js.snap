exports[`test Link changes the class when hovered 1`] = `
<button
  className="btn btn-success"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Search
</button>
`;

exports[`test Link changes the class when hovered 2`] = `
<button
  className="btn btn-info"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Search
</button>
`;

exports[`test Link changes the class when hovered 3`] = `
<button
  className="btn btn-success"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Search
</button>
`;

exports[`test Link changes the class when hovered 4`] = `
<button
  className="btn btn-success"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Search
</button>
`;

exports[`test Link changes the class when hovered 5`] = `
<button
  className="btn btn-success"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Search
</button>
`;
